### PR TITLE
feat: add text extraction from pdf and image

### PR DESCRIPTION
**Pull request by mikhail-zapolskyi**
## Why was this PR opened?
Completed the task pdf and image to text

## Brief Description
add route /extract/pdf. Using the route frontend can send file in pdf or image format and receive the text back in format
{
text: "Lorem ipsum"
}
Before use it tesseract needs to be installed on device. Please refer to readme file

## Link to Referenced Trello Cards
https://trello.com/c/KtUc3yjO/120-input-from-photo-pdf-doc

### CODE REVIEW

File: backend/requirements.txt [hunk 1]
```diff
@@ -6,4 +6,8 @@ Requests==2.31.0
 uvicorn==0.22.0
 python-multipart
 sqlalchemy
-psycopg2
\ No newline at end of file
+psycopg2
+PyPDF2
+pytesseract
+Pillow
+cv2
```
Comment (maggiechew) [inline backend/requirements.txt:13]: I had an error doing pip install, regarding cv2 (ERROR: Could not find a version that satisfies the requirement cv2 (from versions:none) ERROR: No matching distribution found for cv2.)
Some googling suggests that some different versions may be required, e.g. opencv-python?

File: package.json [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-  }
+	"name": "monorepo",
+	"version": "1.0.0",
+	"workspaces": [
+		"apps/*",
+		"packages/*"
+	],
+	"scripts": {
+		"start:app": "node findIP && npm run start --workspace=learning-buddy",
+		"start:backend": "python3 backend/main.py",
```
Comment (maggiechew) [inline package.json:10]: This change caused errors for me

File: backend/requirements.txt
Comment (maggiechew) [inline backend/requirements.txt:1]: It's weird, my git graph isn't working. Can you tell me where these versions for the imports on lines 1-6 came from? It doesn't appear you added them, did they come from branching off of main?

**Review by maggiechew — DISMISSED** [review]
Unfortunately, with the errors I have with cv2 I can't run this. I can test it out once we have that sorted!

File: backend/app/routers/sendText.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
+from fastapi import UploadFile, APIRouter, File
+from ..utils.extractTextPdf import extract_text_pdf
+from ..utils.extractTextImage import extract_text_image
+
+router = APIRouter(
+    prefix='/extract',
+    tags=['exctract']
```
Comment (maggiechew) [inline backend/app/routers/sendText.py:7]: Spelling error :) "extract"

File: backend/requirements.txt (hunk 1 above)
Comment (mikhail-zapolskyi) [reply backend/requirements.txt]: opencv-python package needs to be install. fixed in new update

File: package.json (hunk 2 above)
Comment (mikhail-zapolskyi) [reply package.json:10]: fixed in new update

File: backend/requirements.txt
Comment (mikhail-zapolskyi) [reply backend/requirements.txt:1]: I am not sure.

File: backend/app/routers/sendText.py (hunk 3 above)
Comment (mikhail-zapolskyi) [reply backend/app/routers/sendText.py]: fixed in new update.

**Review by huynhtk80 — APPROVED** [review]
Overall looks good.  Main issue it word document can't parse from tables, but that might be outside of the MVP, unless there is a quick fix.  Just make sure product owner knows.

Nice to have, if no text is found return error of some type.

**Pull request merged**